packages/atomic/src/components/commerce/atomic-commerce-query-error/atomic-commerce-query-error.spec.ts

```diff
@@ -0,0 +1,328 @@
+    it('should render the button with the correct part', async () => {
+      const {locators} = await renderQueryError();
+
+      expect(locators.showMoreInfoButton).toHaveAttribute(
+        'part',
+        'more-info-btn'
+      );
+    });
+
+    it('should render the button with the correct text', async () => {
+      const {locators} = await renderQueryError();
+
+      expect(locators.showMoreInfoButton).toHaveTextContent('Learn more');
+    });
+
+    it('should toggle the show more info state when clicked', async () => {
+      const {locators} = await renderQueryError();
+      expect(locators.errorDetails).toBeNull();
+
+      await userEvent.click(locators.showMoreInfoButton!);
+
+      expect(locators.errorDetails).toBeVisible();
+    });
+
+    describe('when the show more info button is clicked', () => {
+      let showMoreInfoButton: HTMLButtonElement;
+      let errorDetails: HTMLPreElement;
+      beforeEach(async () => {
+        const {locators} = await renderQueryError();
+        showMoreInfoButton = locators.showMoreInfoButton as HTMLButtonElement;
+        await userEvent.click(showMoreInfoButton);
+        errorDetails = locators.errorDetails as HTMLPreElement;
+      });
+
+      it('should display the error details with the correct part', async () => {
+        expect(errorDetails).toHaveAttribute('part', 'error-info');
+      });
+
+      it('should display the stringified error details', async () => {
+        expect(errorDetails.querySelector('code')?.textContent).toBe(
+          JSON.stringify(mockError, null, 2)
+        );
+      });
+
+      it('should hide the error details when clicked again', async () => {
+        await userEvent.click(showMoreInfoButton);
+        expect(errorDetails).not.toBeVisible();
+      });
+    });
+  });
+
+  describe('when the doc link is displayed', () => {
+    let locators: any;
+    beforeEach(async () => {
+      const {locators: renderedLocators} = await renderQueryError({
+        error: {
+          type: 'NoEndpointsException',
+          statusCode: 500,
+          message: 'No endpoints available',
+        },
+      });
+      locators = renderedLocators;
+    });
+
+    it('should render the doc link with the correct part', async () => {
+      expect(locators.docLink).toHaveAttribute('part', 'doc-link');
+    });
+
+    it('should render the doc link with the correct text', async () => {
+      expect(locators.docLink).toHaveTextContent('Coveo Online Help');
+    });
+  });
+
+  describe('when the error type is "Disconnected"', () => {
+    let locators: any;
+
+    beforeEach(async () => {
+      const mockError = {
+        type: 'Disconnected',
+        statusCode: 500,
+        message: 'Disconnected from the server',
+      };
+      ({locators} = await renderQueryError({error: mockError}));
+    });
+
+    it('should display the correct icon', () => {
+      expect(locators.icon).toHaveAttribute('icon', NoConnection);
```

Comment:
I don't think this is relevant in this test. Should probably be tested in query-error/icon.spec.ts